libs/ledger-live-common/src/families/elrond/api/apiCalls.ts

```diff
@@ -52,77 +71,144 @@ export default class ElrondApi {
             erd_min_gas_limit: gasLimit,
             erd_min_gas_price: gasPrice,
             erd_gas_per_data_byte: gasPerByte,
+            erd_gas_price_modifier: gasPriceModifier,
           },
         },
       },
     } = await network({
       method: "GET",
       url: `${this.API_URL}/network/config`,
     });
+
     return {
-      chainId,
+      chainID: chainId,
       denomination,
       gasLimit,
       gasPrice,
       gasPerByte,
+      gasPriceModifier,
     };
   }
 
-  async submit({ operation, signature }) {
-    const { chainId, gasLimit, gasPrice } = await this.getNetworkConfig();
-    const {
-      senders: [sender],
-      recipients: [receiver],
-      value,
-      transactionSequenceNumber: nonce,
-    } = operation;
+  async submit(signedOperation: SignedOperation): Promise<string> {
+    const transaction = {
+      ...signedOperation.signatureRaw,
+      signature: signedOperation.signature,
+    };
+
     const {
       data: {
         data: { txHash: hash },
       },
     } = await network({
       method: "POST",
       url: `${this.API_URL}/transaction/send`,
-      data: {
-        nonce,
-        value,
-        receiver,
-        sender,
-        gasPrice,
-        gasLimit,
-        chainID: chainId,
-        signature,
-        ...HASH_TRANSACTION,
-      },
+      data: transaction,
     });
-    return {
-      hash,
-    };
+
+    return hash;
   }
 
-  async getHistory(addr: string, startAt: number) {
+  async getHistory(
+    addr: string,
+    startAt: number
+  ): Promise<ElrondApiTransaction[]> {
     const { data: transactionsCount } = await network({
       method: "GET",
-      url: `${this.API_URL}/transactions/count?condition=should&sender=${addr}&receiver=${addr}&after=${startAt}`,
+      url: `${this.API_URL}/accounts/${addr}/transactions/count?after=${startAt}`,
     });
 
-    let allTransactions: any[] = [];
+    let allTransactions: ElrondApiTransaction[] = [];
     let from = 0;
-    while (from <= transactionsCount) {
+    while (from < transactionsCount) {
```

Comment:
Not sure about this one, but don't you want to use this same condition in `getESDTTokensForAddress` ?

Reply:
Added, for consistency.